tests/tests_pytorch/checkpointing/test_model_checkpoint.py

```diff
@@ -1127,7 +1127,7 @@ def __init__(self, hparams):
         assert isinstance(ckpt[model.CHECKPOINT_HYPER_PARAMS_KEY], Container)
     else:
         # make sure it's not AttributeDict
-        assert type(ckpt[model.CHECKPOINT_HYPER_PARAMS_KEY]) is dict
```

Comment:
I suggest reverting all or most of the isinstance changes. For instance here we want to check that it's exactly a dict and not a subclass

These changes are not equivalent